### DISCUSSION TITLE
Improvement of Examples

### DISCUSSION THREAD

**Discussion by termi-official** [Ferrite-FEM/Ferrite.jl#372] [Ideas]
Hi everyone,

just to get everyone on the same page I want to shortly discuss the purpose of the examples in Ferrite and how to improve here.

In the current state example seem to have three primary purposes. 
1. Showcasing functionality
2. Integration testing
3. Starting point for new users to derive their codes

While I like having the example (not wanting to diminish the developers efforts) I believe we should improve and standardize the examples to make them easier followable for new users.

My suggestion to improve the first point above is, that each and every example should...
1. start with a very short "What do learn here" section to summarize what the reader can expect to take away methodogically from the example
2. have hints about which ones of the other examples could be read to better understand of the current example, so we could keep advanced examples concise
3. provide sufficient mathematical theory (i.e. give and close all PDEs with the corresponding BCs and major points of the utilized discretization) to follow the implementation - currently some examples can be hard to follow (see e.g. #369 )

Regarding integration testing, maybe it can be useful to compute the examples with smaller meshes and time domains than the ones showcased? Currently the number of examples start to grow, but some of the examples already take quite some amount of time . 1:30 h is too much for such a small code.

**Comment by khorrami1** [comment]
Yes, I completely agree with you. For example, this example [Ginzburg-Landau model](https://ferrite-fem.github.io/Ferrite.jl/dev/examples/landau/) needs to be clarified what the problem is and shows the equations and the corresponding PDE.

**Comment by Avina-cK** [comment]
I'm a beginner contributor and I completely agree on keeping advanced examples concise. However, would it be alright to add collapsible sections explaining derivations? 
For example, in [transient_heat_equation](https://ferrite-fem.github.io/Ferrite.jl/stable/tutorials/transient_heat_equation/#), may I add collapsible sections deriving the semidiscrete weak form and the assembly of discrete operators?

**Reply by termi-official** [reply]
> However, would it be alright to add collapsible sections explaining derivations?

Yes, that would be indeed a very nice contribution! The only things which we require for such contributions are that they should be as correct and internally consistent as possible (and note that there might be some faulty things lingering in the tutorials which we are not aware of). Of course, we try to support contributors here to accomplish this as good as we can.

> For example, in [transient_heat_equation](https://ferrite-fem.github.io/Ferrite.jl/stable/tutorials/transient_heat_equation/#), may I add collapsible sections deriving the semidiscrete weak form and the assembly of discrete operators?

We have the derivation of the "difficult" part for a heat equation derivation given in the introduction here https://ferrite-fem.github.io/Ferrite.jl/stable/topics/fe_intro/ . If you feel like we should add a reference to the transient heat tutorial (and do the same for the stationary heat tutorial), then can remind the reader that this thing exists. Is that what you were having in mind, or what else would you like to see in/contribute to the transient heat tutorial?

**Reply by Avina-cK** [reply]
I agree, I think a reference to the fe_intro page would be good.
I shall include that and change the notation of the test function to be consistent with that in fe_intro page (i.e. v -> δu).

**Comment by Avina-cK** [comment]
As mentioned recently, perhaps we could change the notation of test functions from δu to φ? 
This would remove possible confusion with trial functions u.

**Reply by termi-official** [reply]
Not sure if we should do this for all tutorials, as I think there is some value to it in some of the tutorials. Essentially the difference in notation here comes from which method you apply. If you come from a Ritz-Galerkin perspective (as for example in the hyperleasticity example or in the gallery in ), then you start from some total potential, e.g. $\Pi(u) = \int_\Omega ...$ for which you want to find a $u$ such that the variation of this potential vanishes. In the easiest case this simply corresponds to finding a state of your system which minimized the total energy.

However, as Fredrik also mentioned in the PR, not every PDE can be expressed in terms of energy minimization. For these we start from the PDE in strong form (e.g. from some physical principle like the momentum balance). Here we now either apply a Bubnov-Galerkin or a Petrov-Galerkin approach, where we deal with the system in its weak form.

Now, for energy minimization problems, applying the Bubnov-Galerkin approach or the Ritz-Galerkin approach typically results in the same discrete problem to solve in the end. Does that help with understanding the notation better?
